### PR TITLE
[RLlib] [Connectors] Fix test_io failures when connectors enabled.

### PR DESCRIPTION
**Pull request by avnishn**
Fix in the title

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/offline/json_reader.py [hunk 1]
```diff
@@ -307,6 +308,11 @@ def _postprocess_if_needed(self, batch: SampleBatchType) -> SampleBatchType:
         if not self.ioctx.config.get("postprocess_inputs"):
             return batch
 
+        if isinstance(batch, MultiAgentBatch) and set(batch.policy_batches.keys()) == {
+            DEFAULT_POLICY_ID
```
Comment (gjoliver) [inline rllib/offline/json_reader.py:312]: you don't have to check the only policy key here. convert_ma_batch_to_sample_batch() handles that already.

File: rllib/tests/test_io.py [hunk 2]
```diff
@@ -146,7 +150,8 @@ def test_agent_input_postprocessing_enabled(self):
                         del data["advantages"]
                         if "value_targets" in data:
                             del data["value_targets"]
-                        out.append(data)
+                        data = _to_json(data, [])
+                        out.append(json.loads(data))
```
Comment (gjoliver) [inline rllib/tests/test_io.py:154]: wait, why do we ``_to_json()`` then ``json.loads()`` immediately after?

File: rllib/offline/json_reader.py (hunk 1 above)
Comment (avnishn) [reply rllib/offline/json_reader.py:312]: convert_ma_batch_to_sample_batch throws an error if the ma_batch you pass has more than 1 policy id inside of it. Hence why I added this if statement.

File: rllib/tests/test_io.py (hunk 2 above)
Comment (avnishn) [reply rllib/tests/test_io.py:154]: aha

File: rllib/offline/json_reader.py (hunk 1 above)
Comment (avnishn) [reply rllib/offline/json_reader.py:312]: they don't but someone is using the json reader in a non offline rl setting in this case

**Review by gjoliver — APPROVED** [review]

**Pull request merged**